app/controllers/locations_controller.rb

```diff
@@ -48,6 +48,19 @@ def destroy
     end
   end
 
+  def weeks_exclude_dates
+    start_date = params[:start].to_date
+    end_date = params[:end].to_date - 5.days
+    exclude_dates = []
+    while start_date <= end_date
+      week_days = start_date
+      week_days += 6.days
+      exclude_dates << "#{start_date} - #{week_days}"
+      start_date = week_days
```

Comment:
Once `start_date = week_days`, it is no longer the start_date of the program year (which presumably was the input), so this is confusing naming. How about:

```
workshop_start_date = start_date
while workshop_start_date <= end_date
  workshop_end_date = workshop_start_date + 5.days # 5-Day Workshops
  exclude_dates << "#{workshop_start_date} - #{workshop_end_date}"
  workshop_start_date += 7.days
end